### PR TITLE
Resolves follow-up tasks to signal-based component editing

### PR DESCRIPTION
**Pull request by johnclary**
Fixes [#7136](https://github.com/cityofaustin/atd-data-tech/issues/7136).

- Source signal component definitions from Moped DB
-  Unifies input props for SignalAutoComplete and SignalComponentAutocomplete components
- Addresses a few linter warnings

### CODE REVIEW

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js [hunk 1]
```diff
@@ -455,7 +455,7 @@ const NewProjectMap = ({
         width="100%"
         height="60vh"
         interactiveLayerIds={
-          !isDrawing && !isSignalComponent && getEditMapInteractiveIds()
+          !isDrawing && !isSignalComponent ? getEditMapInteractiveIds() : []
```
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectMap.js:458]: suppresses a map gl warning

File: moped-editor/src/views/projects/newProjectView/NewProjectView.js [hunk 2]
```diff
@@ -492,14 +432,21 @@ const NewProjectView = () => {
     // eslint-disable-next-line
   }, [saveActionState]);
 
+  if (componentQueryloading) {
+    return <CircularProgress />;
+  }
+  if (componentQueryError) return `Error! ${componentQueryError.message}`;
```
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectView.js:438]: Did not put any thought to loading state and error state renders—just copied what i saw in ProjectSummary view.

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js [hunk 3]
```diff
@@ -1,60 +1,19 @@
...
-import { signalToFeatureCollection } from "src/utils/mapHelpers";
-
+import {
+  filterSignalOptions,
+  useSignalChangeEffect,
+  getSignalOptionLabel,
+  getSignalOptionSelected,
+  useInitialSignalComponentValue,
+  renderSignalInput,
+} from "src/utils/signalComponentHelpers";
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:13]: Unified autocomplete behavior between the two versions of the component (new project vs editing).

**Comment by johnclary** [comment]
Folks, this is _mostly_ re-arrangement of files into the new signalComponentHelpers. Otherwise the most substantial work is in the `[refactor to use component data from db](https://github.com/cityofaustin/atd-moped/commit/5e4ac68a52fa2df6a5c4ac3e54c160c28b9fefe8)` commit.

File: moped-editor/src/utils/signalComponentHelpers.js [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    features: [],
+  };
+  if (signal && signal?.properties && signal?.geometry) {
+    /* 
+    / preserves the feature's previous UUID if it's being edited. we are **not** preserving
+    / any other feature properties when the feature is edited. so, for example, if the user
+    / edits a signal component and the signal geometry in socrata has since changed, the new
+    / geometry will be saved.
+    */
+    const featureUUID = signal?.id || uuidv4();
```
Comment (sergiogcx) [inline moped-editor/src/utils/signalComponentHelpers.js:56]: The entire rearrangement looks amazing! 🤝

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js [hunk 5]
```diff
@@ -94,42 +52,17 @@ const SignalComponentAutocomplete = ({
...
       // this label formatting mirrors the Data Tracker formatting
-      getOptionLabel={option =>
-        option
-          ? `${option.properties.signal_id}: ${option.properties.location_name}`
-          : ""
-      }
+      getOptionLabel={getSignalOptionLabel}
       onChange={(e, signal) => {
-        setSignal(signal);
+        setSignal(signal ? signal : null);
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:60]: 🥇

**Review by sergiogcx — APPROVED** [review]
Total pro! this takes the maps to a whole new level! 🙌 🗺️

**Comment by johnclary** [comment]
thanks @sergiogcx! @amenity there's no UI change here, so i'm merging :)

**Pull request merged**